### PR TITLE
adaption for CogVideoX1.5

### PR DESCRIPTION
**Pull request by jiashenggu**
add CogVideoX1.5 rope that considers patch_size_t
https://github.com/huggingface/diffusers/blob/2e86a3f0235cb41b212417d84b9c2cd46d8c1297/src/diffusers/pipelines/cogvideo/pipeline_cogvideox.py#L436
https://github.com/huggingface/diffusers/blob/2e86a3f0235cb41b212417d84b9c2cd46d8c1297/src/diffusers/pipelines/cogvideo/pipeline_cogvideox.py#L663
@a-r-r-o-w

### CODE REVIEW

File: training/args.py [hunk 1]
```diff
@@ -78,6 +78,7 @@ def _get_dataset_args(parser: argparse.ArgumentParser) -> None:
         nargs="+",
         type=int,
         default=[49],
+        help="CogVideoX1.5 need to guaranteed that ((num_frames - 1) // self.vae_scale_factor_temporal + 1) % patch_size_t != 0, such as 53"
```
Comment (sayakpaul) [inline training/args.py:81]:
```suggestion
        help="CogVideoX1.5 need to guarantee that ((num_frames - 1) // self.vae_scale_factor_temporal + 1) % patch_size_t != 0, such as 53"
```

File: training/cogvideox_image_to_video_lora.py [hunk 2]
```diff
@@ -787,6 +787,7 @@ def load_model_hook(models, input_dir):
                         num_frames=num_frames,
                         vae_scale_factor_spatial=VAE_SCALE_FACTOR_SPATIAL,
                         patch_size=model_config.patch_size,
+                        patch_size_t=model_config.patch_size_t if hasattr(model_config, "patch_size_t") else None,
```
Comment (sayakpaul) [inline training/cogvideox_image_to_video_lora.py:790]: Beautiful!

File: training/utils.py [hunk 3]
```diff
@@ -207,14 +208,28 @@ def prepare_rotary_positional_embeddings(
...
+        base_num_frames = (num_frames + patch_size_t - 1) // patch_size_t
+
+        freqs_cos, freqs_sin = get_3d_rotary_pos_embed(
+            embed_dim=attention_head_dim,
+            crops_coords=None,
+            grid_size=(grid_height, grid_width),
+            temporal_size=base_num_frames,
+            grid_type="slice",
+            max_size=(base_size_height, base_size_width),
+        )
```
Comment (sayakpaul) [inline training/utils.py:231]: Do we wanna collate this into a mini function named `get_position_frequencies()`?

**Review by sayakpaul — APPROVED** [review]
Beautiful!

Could we also include a note about this support in the README (and perhaps a training command)? @a-r-r-o-w WDYT?

File: training/utils.py (hunk 3 above)
Comment (jiashenggu) [reply training/utils.py:231]: https://github.com/huggingface/diffusers/blob/2e86a3f0235cb41b212417d84b9c2cd46d8c1297/src/diffusers/pipelines/cogvideo/pipeline_cogvideox.py#L436
I refer this

**Comment by jiashenggu** [comment]
btw, I find an error when I run train_text_to_video_sft.sh without deepspeed, this [pr](https://github.com/a-r-r-o-w/cogvideox-factory/pull/84) solve it

**Review by a-r-r-o-w — APPROVED** [review]
tysm!

**Pull request merged**